pkg/conversion/capi2mapi/powervs.go

```diff
@@ -0,0 +1,305 @@
+	if m.machineSet == nil || m.template == nil || m.powerVSCluster == nil || m.machineAndPowerVSMachineAndPowerVSCluster == nil {
+		return nil, nil, errCAPIMachineSetPowerVSMachineTemplatePowerVSClusterCannotBeNil
+	}
+
+	var (
+		errs     []error
+		warnings []string
+	)
+
+	// Run the full ToMachine conversion so that we can check for
+	// any Machine level conversion errors in the spec translation.
+	mapiPowerVSMachine, warn, err := m.ToMachine()
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	warnings = append(warnings, warn...)
+
+	mapiMachineSet, err := fromCAPIMachineSetToMAPIMachineSet(m.machineSet)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	mapiMachineSet.Spec.Template.Spec = mapiPowerVSMachine.Spec
+
+	// Copy the labels and annotations from the Machine to the template.
+	mapiMachineSet.Spec.Template.ObjectMeta.Annotations = mapiPowerVSMachine.ObjectMeta.Annotations
+	mapiMachineSet.Spec.Template.ObjectMeta.Labels = mapiPowerVSMachine.ObjectMeta.Labels
+
+	if len(errs) > 0 {
+		return nil, warnings, utilerrors.NewAggregate(errs)
+	}
+
+	return mapiMachineSet, warnings, nil
+}
+
+// toProviderSpec converts a capi2mapi machineAndPowerVSMachineAndPowerVSCluster into a MAPI PowerVSMachineProviderConfig.
+func (m machineAndPowerVSMachineAndPowerVSCluster) toProviderSpec() (*mapiv1.PowerVSMachineProviderConfig, field.ErrorList) {
+	errs := field.ErrorList{}
+
+	fldPath := field.NewPath("spec")
+
+	serviceInstance, err := convertPowerVSServiceInstanceToMAPI(fldPath.Child("serviceInstance"), m.powerVSMachine.Spec.ServiceInstanceID, m.powerVSMachine.Spec.ServiceInstance)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	image, err := convertPowerVSImageToMAPI(fldPath.Child("image"), m.powerVSMachine.Spec.Image, m.powerVSMachine.Spec.ImageRef)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	network, err := convertPowerVSNetworkToMAPI(fldPath.Child("network"), m.powerVSMachine.Spec.Network)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	mapiProviderConfig := mapiv1.PowerVSMachineProviderConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PowerVSMachineProviderConfig",
+			APIVersion: "machine.openshift.io/v1",
+		},
+		ServiceInstance: serviceInstance,
+		Image:           image,
+		Network:         network,
+		KeyPairName:     m.powerVSMachine.Spec.SSHKey,
+		SystemType:      m.powerVSMachine.Spec.SystemType,
+		ProcessorType:   mapiv1.PowerVSProcessorType(m.powerVSMachine.Spec.ProcessorType),
+		Processors:      m.powerVSMachine.Spec.Processors,
+		MemoryGiB:       m.powerVSMachine.Spec.MemoryGiB,
+		//CredentialsSecret:
+		//LoadBalancers: TODO(MULTIARCH-5041): Not supported for workers.
+	}
+
+	userDataSecretName := ptr.Deref(m.machine.Spec.Bootstrap.DataSecretName, "")
+	if userDataSecretName != "" {
+		mapiProviderConfig.UserDataSecret = &mapiv1.PowerVSSecretReference{
+			Name: userDataSecretName,
+		}
+	}
+
```

Comment:
Are there any fields that exist in the CAPI machine template, that are not consumer as part of the above conversion to a MAPI provider spec?